source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation_data.py

```diff
@@ -287,15 +319,46 @@ def body_state_w(self):
             self._body_state_w.timestamp = self._sim_timestamp
         return self._body_state_w.data
 
+    @property
+    def body_state_link_w(self):
+        """State of all bodies `[pos, quat, lin_vel, ang_vel]` in simulation world frame.
+        Shape is (num_instances, num_bodies, 13).
+
+        The position, quaternion, and linear/angular velocity are of the body's link frame relative to the world.
+        """
+        state = self.body_state_w.clone()
+        quat = state[..., 3:7]
+        # adjust linear velocity to link
+        state[..., 7:10] += torch.linalg.cross(
+            state[..., 10:13], math_utils.quat_rotate(quat, -self._com_pos_b), dim=-1
+        )
+        return state
+
+    @property
+    def body_state_com_w(self):
+        """State of all bodies `[pos, quat, lin_vel, ang_vel]` in simulation world frame.
+        Shape is (num_instances, num_bodies, 13).
+
+        The position, quaternion, and linear/angular velocity are of the body's center of mass frame relative to the
+        world. Center of mass frame is assumed to be the same orientation as the link rather than the orientation of the
+        principle inertia.
+        """
+        state = self.body_state_w.clone()
+        quat = state[..., 3:7]
+        # adjust position to center of mass
```

Comment:
Is the orientation kept the same as the link? Doens't PhysX report the COM as the principle axis direction? This behavior is then different from users doing "get_coms" call?

Reply:
yeah the current implementation will return a different frame than the get_coms. I am not sure which is more intuitive, but for consistency I could change it to the principle axis formulation.